indra/newview/llviewerjointattachment.h

```diff
@@ -100,17 +100,14 @@ class LLViewerJointAttachment :
 
     bool    hasChanged(const LLVector3 &pos, const LLQuaternion &rot) const
     {
-        static constexpr F32 SMALL_CHANGE(0.05 * 0.05);
-
-        LLVector3 unit_vec(LLVector3::x_axis * rot);
-        F32 unit_cos(0);
-        if (!mLastTrackedUnit.isNull())
-        {
-            unit_cos = unit_vec * mLastTrackedUnit;
-        }
-        mLastTrackedUnit = unit_vec;
-
-        return (((pos - mLastTrackedPos).magVecSquared() > SMALL_CHANGE) || (unit_cos < 0.9));
+        static constexpr F32 SMALL_CHANGE_DIST2(0.05 * 0.05);
+        static constexpr F32 SMALL_CHANGE_ANGL(0.5);    // about 28.6 degrees
```

Comment:
`0.05` --> `0.05f` to remove implicit conversion from `double` to `float`.

`0.5` --> `0.5f`.  Half a radian is rather large for a "small change" threshold.  Are you sure you want `SMALL_CHANGE_ANGL` to have that value?